app/views/spree/admin/slides/_form.html.erb

```diff
@@ -16,10 +16,11 @@
   </div>
 
   <div class="col-md-12">
-    <%= f.field_container :product_id do %>
-      <%= label_tag :product_id, t(:product) %><br>
-      <%= hidden_field_tag 'slide[product_id]', f.object.product_id.to_s, class: "product_picker fullwidth" %>
-    <% end %>
+  <%= f.field_container :product_id do %>
+    <%= f.label :product_id, t(:product) %><br>
+    <%= f.collection_select(:product_id, Spree::Product.all, :id, :name, { include_blank: Spree.t('match_choices.none') }, { class: 'select2', disabled: cannot?(:edit, Spree::Product) }) %>
```

Comment:
@sushil-truemark we should not be fetching data in view..can you please check how existing solution would have worked?

Your recent code changes has been discarding roles and permissions of products..
and will eventually lead to data leak..

Eg: when multivendor gem is added and vendor should be having access to only the product from their vendor list.. this code would reveal all products..

which could be avoided if we were using pundit scope and checking their roles and permission (I am not sure if spree uses pundit.. but there must be some infrastracture for user's ability and scope)